python/tests/core/ecl/test_pylint.py

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env python
+#  Copyright (C) 2017 Statoil ASA, Norway.
+#
+#  This file is part of ERT - Ensemble based Reservoir Tool.
+#
+#  ERT is free software: you can redistribute it and/or modify
+#  it under the terms of the GNU General Public License as published by
+#  the Free Software Foundation, either version 3 of the License, or
+#  (at your option) any later version.
+#
+#  ERT is distributed in the hope that it will be useful, but WITHOUT ANY
+#  WARRANTY; without even the implied warranty of MERCHANTABILITY or
+#  FITNESS FOR A PARTICULAR PURPOSE.
+#
+#  See the GNU General Public License at <http://www.gnu.org/licenses/gpl.html>
+#  for more details.
+
+import fnmatch
+import os
+import unittest
+from pylint import epylint as lint
+from ert.test import ExtendedTestCase
+
+class CodeTestCase(ExtendedTestCase):
+    """This class is a test case for linting."""
+
+    LINT_ARGS = ['-d', 'R,C,W'] + \
+                ['--extension-pkg-whitelist=numpy']
+
+    def _get_lintable_files(self, paths, whitelist=None):
+        """Recursively traverses all folders in paths for *.py files"""
+        matches = []
+        for folder in paths:
+            for root, _, filenames in os.walk(folder):
+                for filename in fnmatch.filter(filenames, '*.py'):
+                    if filename not in whitelist:
+                        matches.append(os.path.join(root, filename))
+        return matches
+
+    def test_linting(self):
+        paths = ['ert/ecl', 'ert/geo', 'ert/util', 'ert/well']  # TODO should be ert, cwrap, tests
+        white = ['ecl_kw.py', 'ecl_type.py', 'ecl_sum.py', 'ecl_grid.py', 'ecl_npv.py']  # TODO fix issues and remove
```

Comment:
It's a _whitelist_ in pylint's terms: the package in question is allowed to have errors.